test/per_file_coverage.sh

```diff
@@ -70,6 +70,9 @@ declare -a KNOWN_LOW_COVERAGE=(
 "source/extensions/io_socket:96.2"
 "source/extensions/io_socket/user_space:96.2"
 "source/extensions/network/dns_resolver/getaddrinfo:96.3"
+"source/extensions/path/rewrite:88.1"
+"source/extensions/path/:96.1"
+"source/extensions/path/rewrite/uri_template:88.1"
```

Comment:
Please increase test coverage so that we can use standard limits for this new code.

Reply:
It is a very small file. We're missing 4 lines of internal errors.
https://storage.googleapis.com/envoy-pr/23d73fb/coverage/source/extensions/path/rewrite/uri_template/uri_template_rewrite.cc.gcov.html

Reply:
There are 3 lines added to per_file_coverage.sh in this pr:

"source/extensions/path/rewrite:88.1"
"source/extensions/path/:96.1"
"source/extensions/path/rewrite/uri_template:88.1"

We previously talked about uri_template_rewrite.cc over chat and when we did you were going to investigate adding test coverage to force errors. "It's three lines of error paths. Will try to force them, 2/3 depend on other functions failing though so it might be hard" What were the results of that investigation?

In any case, what about the other two entries which have lower coverage?

Reply:
The other two are the directories directly above uri_template_rewrite.cc . I've added a test for 1/4 paths. Removed 2/4 as it is dead code bc class can't be created without that validation. The last one is in case the RE2 library fails. We can likely just remove that. Thoughts?